nucypher/characters/lawful.py

```diff
@@ -683,61 +710,59 @@ def get_treasure_map_from_known_ursulas(self, network_middleware, map_identifier
 
     def work_orders_for_capsules(self,
                                  *capsules,
-                                 alice_verifying_key: PublicKey,
-                                 treasure_map: 'TreasureMap' = None,
+                                 label: bytes,
+                                 treasure_map: 'TreasureMap',
+                                 relayer_verifying_key: PublicKey,
+                                 alice_verifying_key: Optional[PublicKey] = None,
                                  num_ursulas: int = None,
                                  ) -> Tuple[Dict[ChecksumAddress, 'WorkOrder'], Dict['Capsule', 'WorkOrder']]:
 
-        from nucypher.policy.collections import WorkOrder  # Prevent circular import
+        from nucypher.policy.orders import WorkOrder
 
-        if treasure_map:
-            treasure_map_to_use = treasure_map
-        else:
-            try:
-                treasure_map_to_use = self.treasure_maps[map_id]
-            except KeyError:
-                raise KeyError(f"Bob doesn't have the TreasureMap {map_id}; can't generate work orders.")
+        if not alice_verifying_key:
+            # Assume the policy publisher is the same as the KFrag generator by default.
+            alice_verifying_key = relayer_verifying_key
```

Comment:
I might be missing something, but this seems backward to me i.e. shouldn't alice_verifying_key should always be provided, and the relayer_verifying_key should be optional...? and then we get:

```python
if not relayer_verifying_key:
    relayer_verifying_key = alice_verifying_key
```

The KFrag generator would be alice_verifying_key, which is always the case, and then the publisher i.e. relayer could either be alice or a separate entity...

Reply:
✔️ Adding more keys is definitely a trade-off between API flexibility and complexity here.